lib/ansible/modules/cloud/azure/azure_rm_webapp.py

```diff
@@ -0,0 +1,930 @@
+                type='bool'
+            ),
+            ttl_in_seconds=dict(
+                type='int'
+            ),
+            app_settings=dict(
+                type='dict'
+            ),
+            purge_app_settings=dict(
+                type='bool',
+                default=False
+            ),
+            state=dict(
+                type='str',
+                default='present',
+                choices=['present', 'absent']
+            )
+        )
+
+        self.resource_group = None
+        self.name = None
+        self.location = None
+
+        # update in create_or_update as parameters
+        self.client_affinity_enabled = True
+        self.force_dns_registration = None
+        self.skip_dns_registration = None
+        self.skip_custom_domain_verification = None
+        self.ttl_in_seconds = None
+        self.https_only = None
+
+        self.tags = None
+
+        # site config, e.g app settings, ssl
+        self.site_config = dict()
+        self.app_settings = dict()
+        self.app_settings_strDic = None
+
+        # app service plan
+        self.plan = None
+        # siteSourceControl
+        self.deployment_source = dict()
+
+        # site, used at level creation, or update. e.g windows/linux, client_affinity etc first level args
+        self.site = None
+
+        # property for internal usage, not used for sdk
+        self.container_settings = None
+
+        self.purge_app_settings = False
+
+        self.results = dict(changed=False)
+        self.state = None
+        self.to_do = Actions.NoAction
+
+        # set site_config value from kwargs
+        self.site_config_properties = ["net_framework_version",
+                                       "java_version",
+                                       "php_version",
+                                       "python_version",
+                                       "linux_fx_version",
+                                       "scm_type"]
+
+        # updatable_properties
+        self.updatable_properties = ["client_affinity_enabled",
+                                     "force_dns_registration",
+                                     "https_only",
+                                     "skip_custom_domain_verification",
+                                     "ttl_in_seconds"]
+
+        super(AzureRMWebApps, self).__init__(derived_arg_spec=self.module_arg_spec,
+                                             supports_check_mode=True,
+                                             supports_tags=True)
+
+    def exec_module(self, **kwargs):
+        """Main module execution method"""
+
+        setattr(self, 'tags', kwargs['tags'])
+
+        for key in list(self.module_arg_spec.keys()):
+            if hasattr(self, key):
+                setattr(self, key, kwargs[key])
+            elif kwargs[key] is not None:
+                if key in self.site_config_properties:
+
+                    self.site_config[key] = kwargs[key]
```

Comment:
we can add one item in spec `site_config=dict(type='dict', default=dict())`, then user can add there configuration in the site_config dict

Reply:
consideration here is to move the most frequently used configuration like app_setting, container_setting etc as module arguments, so that user don't need to fill in a deep site_config. check [here](https://docs.microsoft.com/en-us/rest/api/appservice/webapps/createorupdate#siteconfig) to see how big a site config is. if use a name `site_config` in module parameter, it give user some hints it will be something like that.